### PR TITLE
Add more Admin logic

### PR DESCRIPTION
**Pull request by Danielsio**
Added more methods to admin controller and admin Engine.
1. method that will make specific user admin
2. method that qeury all users

### CODE REVIEW

**Comment by Danielsio** [comment]
Code should be Reviewed cautiously before getting mergerd.

File: src/main/java/com/example/planit/engine/AdminEngine.java [hunk 1]
```diff
@@ -146,4 +147,59 @@ public DTOcoursesResponseToController getCourseFromDB(String courseId, String su
...
+                return new DTOusersResponseToController(false, Constants.ERROR_USER_NOT_FOUND, HttpStatus.BAD_REQUEST);
+            }
+
+            User maybeAdminUser = maybeUser.get();
+
+            if (!maybeAdminUser.isAdmin()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_UNAUTHORIZED_USER, HttpStatus.UNAUTHORIZED);
+            }
+
+            // Check if course exists in database
```
Comment (oshriFadlon1) [inline src/main/java/com/example/planit/engine/AdminEngine.java]: // user not course

Comment (Danielsio) [reply src/main/java/com/example/planit/engine/AdminEngine.java]: nice catch

**Comment by Danielsio** [comment]
When you are done reviewing the code changes, please leave a comment, or approve the changes, so I would know that someone is doing something. Thanks.

File: src/main/java/com/example/planit/engine/AdminEngine.java [hunk 2]
```diff
@@ -146,4 +151,67 @@ public DTOcoursesResponseToController getCourseFromDB(String courseId, String su
...
+            User maybeAdminUser = maybeUser.get();
+
+            if (!maybeAdminUser.isAdmin()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_UNAUTHORIZED_USER, HttpStatus.UNAUTHORIZED);
+            }
+
+            // Check if user exists in database
+            Optional<User> maybeExistingUser = userRepo.findUserBySubjectID(userSubId);
+            if (maybeExistingUser.isEmpty()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_USER_NOT_FOUND, HttpStatus.BAD_REQUEST);
```
Comment (itayf9) [inline src/main/java/com/example/planit/engine/AdminEngine.java:202]: The constant : `Constants.ERROR_USER_NOT_FOUND` is returned in the response both when the admin user is not in the DB (line 196)  and when the user to be converted is not in the DB (line 202).

**Review by itayf9 — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/planit/engine/AdminEngine.java (hunk 2 above)
Comment (itayf9) [reply src/main/java/com/example/planit/engine/AdminEngine.java:202]: The issue #18 will fix that problem, which is in a greater scale, in the future. So I think this PR can be merged.

**Review by itayf9 — APPROVED** [review]

**Pull request merged**